Assignment 2.Rmd

```diff
@@ -81,19 +86,82 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
 
 plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
 
-````
+```
 
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+```{r}
+E.STD <- EDGE %>% group_by(to) %>% summarise(sum(count))
+E.TCH <- EDGE %>% group_by(from) %>% summarise(sum(count))
+names(E.STD) <- c("id","count")
+names(E.TCH) <- c("id","count")
+vertex <- bind_rows(E.STD, E.TCH)
+
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+plot(g,layout=layout.fruchterman.reingold, vertex.size=VERTEX$count, edge.width=EDGE$count)
+
+```
 
 
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
+```{r}
+library(tidyr)
+library(igraph)
+library(dplyr)
+
+raw_data1 <- read.csv("hudk4050-classes.csv")
+DF <- select(raw_data, Q10, Q1:Q7) #select useful column 
```

Comment:
Did you use data from last year? I did not see where is Q10 or Q1/Q7 in this dataset.

Reply:
<img width="835" alt="Screen Shot 2019-10-24 at 5 35 31 PM" src="https://user-images.githubusercontent.com/45274600/67527336-c142b080-f684-11e9-89d9-d84799181070.png">
I am certain that I download the right data, because I can find my own UNI in the data. This picture is what  "raw.data1" looks like for me.
Thank you very much